File: tmdb-core/src/main/java/dev/reuss/tmdb/domain/watchproviders/model/WatchProviderRegion.java
package dev.reuss.tmdb.domain.watchproviders.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import dev.reuss.tmdb.common.TmdbModel;

/**
 * Region with available TMDB watch provider data.
 *
 * @param iso31661    ISO 3166-1 region code
 * @param englishName English region name
 * @param nativeName  native region name
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record WatchProviderRegion(
        @JsonProperty("iso_3166_1")
        String iso31661,

        @JsonProperty("english_name")
        String englishName,

        @JsonProperty("native_name")
        String nativeName
) implements TmdbModel {
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/domain/watchproviders/model/WatchProviderRegion.kt
package dev.reuss.tmdb.domain.watchproviders.model

import com.fasterxml.jackson.annotation.JsonProperty
import dev.reuss.tmdb.common.TmdbModel

/**
 * Region with available TMDB watch provider data.
 *
 * @property iso31661 ISO 3166-1 region code
 * @property englishName English region name
 * @property nativeName native region name
 */
@JvmRecord
data class WatchProviderRegion(
    @all:JsonProperty("iso_3166_1")
    val iso31661: String?,

    @all:JsonProperty("english_name")
    val englishName: String?,

    @all:JsonProperty("native_name")
    val nativeName: String?
) : TmdbModel